test/unit/BPool.t.sol

```diff
@@ -0,0 +1,3539 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.23;
+
+import {BPool} from 'contracts/BPool.sol';
+import {MockBPool} from 'test/smock/MockBPool.sol';
+
+import {BConst} from 'contracts/BConst.sol';
+import {BMath} from 'contracts/BMath.sol';
+import {IERC20} from 'contracts/BToken.sol';
+import {Test} from 'forge-std/Test.sol';
+import {LibString} from 'solmate/utils/LibString.sol';
+import {Pow} from 'test/utils/Pow.sol';
+import {Utils} from 'test/utils/Utils.sol';
+
+abstract contract BasePoolTest is Test, BConst, Utils, BMath {
+  using LibString for *;
+
+  uint256 public constant TOKENS_AMOUNT = 3;
+
+  MockBPool public bPool;
+  address[TOKENS_AMOUNT] public tokens;
+
+  // Deploy this external contract to perform a try-catch when calling bpow.
+  // If the call fails, it means that the function overflowed, then we reject the fuzzed inputs
+  Pow public pow = new Pow();
+
+  function setUp() public {
+    bPool = new MockBPool();
+
+    // Create fake tokens
+    for (uint256 i = 0; i < tokens.length; i++) {
+      tokens[i] = makeAddr(i.toString());
+    }
+  }
+
+  function _setRandomTokens(uint256 _length) internal returns (address[] memory _tokensToAdd) {
+    _tokensToAdd = new address[](_length);
+    for (uint256 i = 0; i < _length; i++) {
+      _tokensToAdd[i] = makeAddr(i.toString());
+      _setRecord(_tokensToAdd[i], BPool.Record({bound: true, index: i, denorm: 0, balance: 0}));
+    }
+    _setTokens(_tokensToAdd);
+  }
+
+  // TODO: move tokens and this method to Utils.sol
+  function _tokensToMemory() internal view returns (address[] memory _tokens) {
```

Comment:
👀

Reply:
hmmm i guess we could improve it, @0xAustrian wdyt about making this on #52 (+ perhaps the assume no precompiles)